src/utils.c

```diff
@@ -124,7 +127,7 @@ SEXP colnamesInt(SEXP x, SEXP cols, SEXP check_dups) {
     }
     int *icols = INTEGER(ricols);
     for (int i=0; i<nc; i++) {
-      if ((icols[i]>nx) || (icols[i]<1))
+      if (((icols[i]>nx) || (icols[i]<1)) && !Skip_absent)
```

Comment:
nit: I would put the skip_absent check first, then we can avoid one (tiny) calculation when skipping

Reply:
done 👍